### PR TITLE
Modified test, so that all nodes don’t all connect to the first node only

### PR DESCRIPTION
**Pull request by youngjoon-lee**
### Background
Previously in integration tests, all nodes make a connection to only the first node, as described in https://github.com/logos-co/nomos-node/issues/440#issue-1913706550. It should be okay, but some nodes sometimes don't receive messages from the first node, even if the nodes send IWANT requests to the first node. Please see details in https://github.com/logos-co/nomos-node/issues/440. 

### What I've done in this PR
Although I couldn't find the reason why some nodes don't receive messages yet, I modified the test little bit in order to unblock #415 and #426 because I don't think that this behaviour is directly tied to our Nomos implementation (overlay, consensus, mixnet, and so on).
Instead of making all nodes connect to the first node, I made each node connect to the only one preceding node. With this, gossipsub messages could be broadcasted without involving IHAVE/IWANT heartbeats.

This will help us focusing on the core logic and make debugging easier.

### What's next
I'm going to keep debugging to find the reason why IWANT requests sometime don't work, by making a simple libp2p testing environment. I'll keep posting my debugging history in https://github.com/logos-co/nomos-node/issues/440#issuecomment-1736740730.

### CODE REVIEW

**Review by al8n — APPROVED** [review]
👍

File: tests/src/nodes/nomos.rs [hunk 1]
```diff
@@ -182,15 +171,17 @@ impl Node for NomosNode {
...
-                for mut conf in configs {
-                    conf.network
-                        .backend
-                        .initial_peers
-                        .push(listening_addr.clone());
+                configs.swap(0, next_leader_idx);
+
+                let mut nodes = Vec::new();
+                for (i, conf) in configs.clone().iter_mut().enumerate() {
+                    if i > 0 {
```
Comment (danielSanchezQ) [inline tests/src/nodes/nomos.rs:178]: Nitpick, you can actually `configs.clone().iter_mut().enumerate().skip(1)`.

File: tests/src/nodes/nomos.rs [hunk 2]
```diff
@@ -182,15 +171,17 @@ impl Node for NomosNode {
...
+                        let prev_node_port = configs[i - 1].network.backend.inner.port;
+                        let addr = multiaddr!(Ip4([127, 0, 0, 1]), Tcp(prev_node_port));
+                        conf.network.backend.initial_peers.push(addr);
```
Comment (danielSanchezQ) [inline tests/src/nodes/nomos.rs:181]: Just so I understand, we make a chain now?

**Review by zeegomo — COMMENTED** [review]
The `SpawnConfig` parameter was designed for this initially. If we want to change the topology we should account for that, for example by adding the `Chain` variant

File: tests/src/nodes/nomos.rs (hunk 2 above)
Comment (youngjoon-lee) [reply tests/src/nodes/nomos.rs:181]: Yes, right. But, I'll try the way that @zeegomo suggested below: https://github.com/logos-co/nomos-node/pull/442#pullrequestreview-1652771120

**Comment by youngjoon-lee** [comment]
> The `SpawnConfig` parameter was designed for this initially. If we want to change the topology we should account for that, for example by adding the `Chain` variant

Thank you. 
3048ccd: Added `SpawnConfig::Chain` (with some refactorings)

**Review by danielSanchezQ — APPROVED** [review]
🚀

**Comment by youngjoon-lee** [comment]
@zeegomo I'm merging this PR to unblock #415 since I applied your suggestion, but I'll keep debugging the IWANT issue: https://github.com/logos-co/nomos-node/issues/440#issuecomment-1736740730. Please let me know if we need to revisit this PR.

**Pull request merged**